lib/bolt_server/transport_app.rb

```diff
@@ -435,13 +489,42 @@ def make_winrm_target(target_hash)
       end
     end
 
+    # Fetches the list of plans for a project
+    #
+    # @param project_ref [String] the project to fetch the list of plans from
+    get '/project_plans' do
+      in_bolt_project(params['project_ref']) do |context|
+        plans_response = plan_list(context[:pal], context[:config].project.plans).to_json
```

Comment:
Can you add a comment here describing how we are using `context[:config].project.plans` to construct the allowlist and only show those plans?